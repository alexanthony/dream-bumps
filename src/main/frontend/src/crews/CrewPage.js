import React, {PropTypes} from 'react';
import {Grid, Cell} from 'material-grid/dist';

import CrewMemberList from './CrewMemberList';
import CrewDetails from './CrewDetails';
import CrewPriceHistory from './CrewPriceHistory';

const CrewPage = ({ params }) => (
  <Grid>
    <Cell col={6}>
      <CrewDetails
        crewId={params.crewId}
      />
    </Cell>
    <Cell col={6}>
      <CrewMemberList
        crewId={params.crewId}
      />
    </Cell>
    <Cell col={12}>
      <CrewPriceHistory
        crewId={params.crewId}
      />
    </Cell>
  </Grid>
);

CrewPage.propTypes = {
  params: PropTypes.shape({
    crewId: PropTypes.string
  })
};

export default CrewPage;
