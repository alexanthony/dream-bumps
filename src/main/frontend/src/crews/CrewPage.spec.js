import React from 'react';
import { shallow } from 'enzyme';
import { CrewPage } from './CrewPage';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const component = shallow(
    <CrewPage
      params={{crewId: '1'}}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});
