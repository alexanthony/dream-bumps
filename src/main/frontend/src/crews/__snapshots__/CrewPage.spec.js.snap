exports[`test Renders properly 1`] = `
<Grid>
  <Cell
    col={6}>
    <Connect(CrewDetails)
      crewId="1" />
  </Cell>
  <Cell
    col={6}>
    <Connect(CrewMemberList)
      crewId="1" />
  </Cell>
  <Cell
    col={12}>
    <Connect(CrewPriceHistory)
      crewId="1" />
  </Cell>
</Grid>
`;
